sklearn/ensemble/tests/test_iforest.py

```diff
@@ -262,14 +262,17 @@ def test_iforest_subsampled_features():
 def test_iforest_average_path_length():
     # It tests non-regression for #8549 which used the wrong formula
     # for average path length, strictly for the integer case
+    # Updated to check average path length when input is <= 2 (issue #11839)
 
     result_one = 2. * (np.log(4.) + np.euler_gamma) - 2. * 4. / 5.
     result_two = 2. * (np.log(998.) + np.euler_gamma) - 2. * 998. / 999.
-    assert_almost_equal(_average_path_length(1), 1., decimal=10)
+    assert _average_path_length(0) == 0.
+    assert _average_path_length(1) == 0.
+    assert _average_path_length(2) == 1.
     assert_almost_equal(_average_path_length(5), result_one, decimal=10)
     assert_almost_equal(_average_path_length(999), result_two, decimal=10)
-    assert_array_almost_equal(_average_path_length(np.array([1, 5, 999])),
-                              [1., result_one, result_two], decimal=10)
+    assert_array_almost_equal(_average_path_length(np.array([1, 2, 5, 999])),
```

Comment:
Since we are changing this line, could we use `assert_allclose`

Reply:
Yes. What's the difference? `assert_allclose` does not check the shapes are the same?

Reply:
all_close use rtol atol instead of decimal. It is just recommended by numpy for consistency:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_array_almost_equal.html